### PR TITLE
Add proactive version compatibility checking for Positron extensions

### PR DESCRIPTION
**Pull request by falbukrek**
Addresses #11322.

Implements version compatibility checking for Positron extensions to provide clear error messages when users attempt to install extensions requiring newer Positron versions.

Previously, when a user tried to install an extension with `engines.positron` set to a version newer than their current Positron installation, they would see a cryptic "Cannot read the extension" error. This was confusing because it suggested the extension file was corrupted rather than incompatible.

This PR adds early validation that checks the `engines.positron` field in extension manifests against the current Positron version before attempting installation. When an incompatible version is detected, users now receive a clear message indicating the required vs. installed Positron version.

### Technical Implementation

**File 1: `abstractExtensionManagementService.ts`**
- Enhanced `positronExtensionCompatibility()` to accept an optional `productService` parameter. Withouth knowing the running version of the product, compatibility cannot be determined.
- Added validation logic that calls `validatePositronExtensionManifest()` when extension has `engines.positron` requirement
- Returns structured compatibility result with clear error messages

**File 2: `extensionManagementService.ts`**
- Updated three installation methods to pass `this.productService` to compatibility checks:
  - `installVSIX()` - for local .vsix file installations
  - `installGalleryExtensions()` - for batch marketplace installs
  - `installFromGallery()` - for single marketplace installs

**File 3: `positronExtensionCompatibility.test.ts` (new)**
- Added comprehensive test suite with 7 test cases covering:
  - Compatible and incompatible version validation
  - Malformed version syntax handling
  - Backward compatibility (extensions without version requirements)
  - Graceful degradation when ProductService unavailable
  - Support for different extension input types

#### Bug Fixes
- Fix misleading "Cannot read the extension" error message when attempting to install extensions requiring newer Positron versions. Users now see clear version compatibility messages. (#11322)

### QA Notes

@:extensions

#### Manual Testing Steps

1. **Test incompatible extension (requires manual setup):**
   - Find or create an extension with `engines.positron` set to a future version (e.g., `"^2026.12.0"`)
   - Attempt to install the extension via VSIX or marketplace
   - Verify you receive a clear error message like: "Extension requires Positron 2026.12.0 but you have [current version]"
   - Verify you DO NOT see "Cannot read the extension" error

2. **Test compatible extension:**
   - Install any extension from the marketplace that either:
     - Has no `engines.positron` field
     - Has `engines.positron` compatible with your version
   - Verify installation succeeds normally

3. **Test blocked extension (regression check):**
   - Attempt to install `ms-python.python` from marketplace
   - Verify you still see the "conflicts with Positron built-in features" message

Test vsix files for steps 1 and 2 are attached: [manual-test-extensions.zip](https://github.com/user-attachments/files/24547266/manual-test-extensions.zip)
Here is a video illustrating the use of these files: 

https://github.com/user-attachments/assets/8db56f2b-e6f0-491d-ba7c-b03ee1de9c42

#### Automated Testing

The new test file `positronExtensionCompatibility.test.ts` contains unit tests that verify:
- Version compatibility validation logic
- Error message formatting
- Backward compatibility with extensions lacking version requirements
- Graceful handling of edge cases

Run the test suite to verify all scenarios pass:
```bash
./scripts/test.sh --run src/vs/platform/extensionManagement/test/common/positronExtensionCompatibility.test
```

### CODE REVIEW

File: src/vs/platform/extensionManagement/common/abstractExtensionManagementService.ts [hunk 1]
```diff
@@ -82,6 +87,30 @@ export function positronExtensionCompatibility(extension: { name: string; publis
...
+	// Check version compatibility for manifests with Positron engine requirements
+	if (productService?.positronVersion) {
+		const manifest = extension as IExtensionManifest;
+		if (manifest.engines?.positron) {
+			const validations = validatePositronExtensionManifest(
+				productService.positronVersion,
+				productService.date,
+				URI.file(''), // extension not yet installed; pass empty URI
+				manifest,
+				false // extensionIsBuiltin is false, because installed extensions are always non-builtin
```
[the remark this answers is not in the document]
Comment (falbukrek) [reply src/vs/platform/extensionManagement/common/abstractExtensionManagementService.ts:100]: Type guard added.
